xarray/backends/api.py

```diff
@@ -664,9 +665,12 @@ def to_netcdf(dataset, path_or_file=None, mode='w', format=None, group=None,
     if not sync:
         return store
```

Comment:
If you look above, we only set `sync = None` if a writer is provided explicitly, which is only used for `open_mfdataset`. This return value should probably be eliminated -- instead we could always return a delayed object for open_mfdataset.

Reply:
@shoyer - did you mean `save_mfdataset` instead of `open_mfdataset`?

Reply:
Yes, my mistake.